operators/tokenizer/bpe_kernels.cc

```diff
@@ -760,99 +743,145 @@ OrtxStatus JsonFastTokenizer::Load(const ort_extensions::bpe::TokenJsonConfig& c
   // re-bind the configuration object
   bpe_conf_ = json_conf_;
 
-  OrtxStatus status;
-  if (tiktoken_){
-    status = bbpe_tokenizer_->Load(vocab,
-                                   merges,
-                                   bpe_conf_.get().GetSpecialTokens().c_str(),
-                                   false);
+  // Check for SPM model
+  json_conf_.spm_model_ = CheckForSpmModel(tok_json);
 
-    module_ifs >> tok_json;
-  } else {
-    ifs >> tok_json;
-    // doesn't work for json with nested objects
-    // auto decoders_node = tok_json.find("/decoder/decoders"_json_pointer);
-    bool has_decoders_node = false;
-    auto decoders_node = tok_json.end();
-    auto decoder_node = tok_json.find("decoder");
-    if (decoder_node != tok_json.end()) {
-      decoders_node = decoder_node->find("decoders");
-      if (decoders_node != decoder_node->end()) {
-        has_decoders_node = true;
+  auto model_node = tok_json.find("model");
+  if (model_node == tok_json.end()) {
+    return OrtxStatus(kOrtxErrorCorruptData, "Failed to get model node from tokenizer.json");
+  }
+
+  bbpe_tokenizer_ = std::make_unique<BpeModel>();
+  OrtxStatus status = bbpe_tokenizer_->Load(*model_node,
+                                            bpe_conf_.get().GetSpecialTokens().c_str(),
+                                            bpe_conf_.get().spm_model_);
+  if (!status.IsOk()) {
+    return status;
+  }
+
+  status = LoadAddedTokens(tok_json, config);
+  if (!status.IsOk()) {
+    return status;
+  }
+
+  add_bos_token_ = config.add_bos_token_;
+  add_eos_token_ = config.add_eos_token_;
+  UpdateTokenAdditionFlags(tok_json, config);
+
+  return status;
+}
+
+OrtxStatus JsonFastTokenizer::LoadTikTokenBase64(const ort_extensions::bpe::TokenJsonConfig& config) {
+  std::string voc_file = config.GetVocabDataFile();
+  std::ifstream ifs = path(voc_file).open();
+  if (!ifs.is_open()) {
+    return OrtxStatus(kOrtxErrorInvalidFile, "Failed to open json file: " + voc_file);
+  }
+
+  // Following vocab and merges only used for tiktoken case but accessed outside scope below
```

Comment:
this comment can now be removed since we have separate methods for loading tiktoken and regular json tokenizer files.